arrow-docs/docs/fx/README.md

```diff
@@ -87,158 +92,75 @@ suspend fun greet(): Unit {
 }
 ```
 
-#### `fx` composition
-
-Side effects can be composed and turned into pure values in `fx` blocks.
-
-#### Turning side effects into pure values with `effect`
-
-`effect` wraps the effect and turns it into a pure value by lifting any `suspend () -> A` user-declared side effect into an `IO<A>` value.
-
-```kotlin:ank:playground
-import arrow.fx.IO
-import arrow.fx.extensions.fx
-//sampleStart
-suspend fun sayHello(): Unit =
-  println("Hello World")
-
-suspend fun sayGoodBye(): Unit =
-  println("Good bye World!")
-
-fun greet(): IO<Nothing, Unit> =
-  IO.fx<Unit> {
-    val pureHello = effect { sayHello() }
-    val pureGoodBye = effect { sayGoodBye() }
-  }
-//sampleEnd
-fun main() {
-  println(greet())
-}
-```
+## Executing effectful programs
 
-When we capture suspended side effects as `IO` values with `effect`, we can pass them around and compose them until we are ready to apply the effects. At this point, nothing has happened because `greet` and `effect` are lazy values.
+The `greet` program is ready to run as soon as the user is ready to commit to an execution strategy that is either `blocking` or `non-blocking`.
+`blocking` execution strategies will block the current thread that's waiting for the program to yield a value, whereas `non-blocking` strategies will immediately return and perform the program's work without blocking the current thread.
 
-#### Applying side effects with `!effect`
+Since both blocking and non-blocking execution scenarios perform side effects, we consider running effects as an `unsafe` operation.
 
-We apply side effects with the operator `!`. Once you purify a side effect with `effect`, you can extract its value in a non-blocking way. `!effect` takes the suspended side effects `sayHello()` and `sayGoodbye()` and ensures the continuation context controls them before they get a chance to be executed. This ensures our effect compositions are pure and referentially transparent, and will only run at the edge.
+Arrow restricts the ability to run programs to extensions of the `Enviroment` type class.
```

Comment:
I understand we'd likely suggest doing it, but does Arrow really **restrict** it? I can still run suspended programs without it can't I? Maybe *"Arrow promotes the ability..."* instead?

Reply:
You're absolutely right. I was trying to emphasize that this is the only type that provides runners, and it should only be used at the edge.

Reply:
Yep, for that I think "Arrow promotes..." is more accurate without losing any meaning.